bouncer/tests/delta_base_ingress.ts

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env -S pnpm tsx
+
+// Test the delta based ingress feature of Solana works as intended.
+// The test will initiate and witness a swap from Solana. It will then restart the engine and ensure
+// that a new swap is not scheduled upon restart. Finally it will kill the engine again, make a deposit
+// while the engine is down and ensure that the swap is started upon restart. It checks that the swap
+// is not an accumulated amount but rather just a delta ingress.
+
+// Args:
+// --bins <path to directory containing node and CFE binaries>.
+// --localnet_init <path to localnet init directory>.
+// --nodes <1 or 3>: The number of nodes running on your localnet. Defaults to 1.
+
+// To run locally:
+// ./tests/delta_base_ingress.ts prebuilt --bins ./../target/debug --localnet_init ./../localnet/init
+// To run in CI:
+// ./tests/delta_base_ingress.ts prebuilt --bins ./../ --localnet_init ./../localnet/init
+
+import yargs from 'yargs';
+import { hideBin } from 'yargs/helpers';
+import { doPerformSwap } from '../shared/perform_swap';
+import { testSwap } from '../shared/swapping';
+import {
+  amountToFineAmount,
+  assetDecimals,
+  executeWithTimeout,
+  observeFetch,
+  sleep,
+} from '../shared/utils';
+import { observeEvent } from '../shared/utils/substrate';
+import { killEngines, startEngines } from '../shared/upgrade_network';
+
+let swapsWitnessed: number = 0;
```

Comment:
move inside the function?

Reply:
already done in the commit pushed almost at the same time  as the comment